packages/lab/src/combo-box-next/useComboBox.tsx

```diff
@@ -1,28 +1,40 @@
-import { FocusEvent, KeyboardEvent, SyntheticEvent, useState } from "react";
+import { FocusEvent, KeyboardEvent, SyntheticEvent } from "react";
 import { useList, UseListProps } from "../list-next/useList";
 import { useComboboxPortal, UseComboBoxPortalProps } from "./useComboboxPortal";
+import { useControlled } from "@salt-ds/core";
 
 interface UseComboBoxProps {
+  value?: string;
+  defaultValue?: string;
   onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
   onFocus?: (event: FocusEvent<HTMLInputElement>) => void;
   onMouseOver?: (event: SyntheticEvent) => void;
   onKeyDown?: (event: KeyboardEvent<HTMLInputElement>) => void;
+  onSelect?: (
+    event: SyntheticEvent<HTMLInputElement>,
+    data: { value?: string }
+  ) => void;
   PortalProps?: UseComboBoxPortalProps;
-  listProps: UseListProps;
+  useListProps: UseListProps;
 }
 
 export const useComboBox = ({
+  value,
+  defaultValue,
   onFocus,
   onBlur,
   onMouseOver,
   onKeyDown,
   PortalProps,
-  listProps,
+  useListProps,
+  onSelect,
 }: UseComboBoxProps) => {
-  const { defaultSelected, ...restListProps } = listProps;
-  const [inputValue, setInputValue] = useState<string | undefined>(
-    defaultSelected
-  );
+  const [inputValue, setInputValue] = useControlled({
+    controlled: value,
+    default: useListProps.defaultSelected || defaultValue,
```

Comment:
🤔 would there be a case where both props defaultSelected and defaultValue are present? if so, is there one that takes precedence over another

Reply:
yes, selected should take precedent